tests/test_database_integration.py

```
@@ -0,0 +1,1024 @@
+            # Nested corruption
+            '{"results": [{"title": null, "authors": {}}]}',
+            
+            # Valid structure but corrupted data
+            '{"results": [{"id": "test", "title": "\\u0000\\u0001Invalid Unicode"}]}'
+        ]
+        
+        # Test corrupted response handling
+        for i, corrupted_response in enumerate(corrupted_responses):
+            # Mock corrupted API response
+            async def corrupted_api_call(query):
+                if i == 0:  # Invalid JSON
```

Ruff (0.11.9)

Function definition does not bind loop variable `i`

(B023)